src/language.cpp

```diff
@@ -35,7 +35,31 @@ TextString::TextString()
 void TextString::Clear()
 {
 	this->name = nullptr;
-	std::fill_n(this->languages, lengthof(this->languages), nullptr);
+	std::fill_n(this->languages, lengthof(this->languages), std::vector<const char*>());
+}
+
+/**
+ * Get the string in the currently selected language.
+ * @param amount The amount for which to return the appropriate plural form.
+ * @return Text of this string in the currently selected language.
+ */
+std::string TextString::GetString(int64 amount) const
+{
+	if (_current_language < 0 || _current_language >= LANGUAGE_COUNT) {
+		return "<out of bounds>";
+	}
+
+	uint plural = _language.GetPluralFormIndex(_current_language, amount);
+	if (plural < this->languages[_current_language].size() && this->languages[_current_language].at(plural) != nullptr) {
+		return this->languages[_current_language].at(plural);
+	}
+
+	plural = _language.GetPluralFormIndex(LANG_EN_GB, amount);
+	if (plural < this->languages[LANG_EN_GB].size() && this->languages[LANG_EN_GB].at(plural) != nullptr) {
+		return this->languages[LANG_EN_GB].at(plural);
+	}
```

Comment:
Any reason why you don't hide this in `this->languages[LANG_EN_GB].getString(amount)` ?

At some point you will need gender as well so this isn't the last thing.